chain/chain/src/flat_storage_resharder.rs

```diff
@@ -188,6 +184,9 @@ impl FlatStorageResharder {
         // Cancel any scheduled, not yet started event.
         self.cancel_scheduled_event();
 
+        let manager = self.runtime.get_flat_storage_manager();
+        manager.split_started();
```

Comment:
I'm think there might be a better way to handle the `split_started` "flagging". In this PR it's an object held by flat storage manager. The downside is that we have to be careful about setting it correctly during restarts and if there are failures, which is already taken care of for `resharding_event` and flat storage status on disk.

Conceptually the "split" is finished when the `resharding_event` is removed and/or when the flat storage status for children enters the catchup phase. 

Although by the usage it doesn't seem we are interested in split being done per se, but rather in a way to stop catchup and handover to state sync.

Would it make sense to read the value of the current flat head from Disk on the fly instead of managing the `split_in_progress` variable?

Reply:
yea good point, added `resharding_catchup_height_reached()` instead that just reads the statuses and finds the lowest height reached so far